ocaml/typing/typeclass.ml

```diff
@@ -1359,10 +1360,15 @@ and class_expr_aux cl_num val_env met_env virt self_scope scl =
               end else
                 match Btype.extract_label name sargs with
                 | Some (l', sarg, _, remaining_sargs) ->
-                    if not optional && Btype.is_optional l' then
-                      Location.prerr_warning sarg.pexp_loc
-                        (Warnings.Nonoptional_label
-                           (Printtyp.string_of_label l));
+                    if not optional && Btype.is_optional l' then (
+                      if Btype.is_position l then
+                        raise
+                          (Error
+                             (sarg.pexp_loc, val_env, Nonoptional_call_pos_label (Printtyp.string_of_label l)))
```

Comment:
The convention in the compiler is to keep lines to 80 characters.

And it might be a small improvement to bind `Printtyp.string_of_label l` to a variable before the inner `if`.

Reply:
Changed! (committed+pushed, `make ci` built locally, but still waiting for ci to finish on github actions)